x/marketmap/keeper/keeper.go

```diff
@@ -59,22 +70,125 @@ func (k Keeper) GetAllTickers(ctx sdk.Context) ([]types.Ticker, error) {
-func (k Keeper) CreateTicker(ctx sdk.Context, ticker types.Ticker) error {
+func (k *Keeper) CreateTicker(ctx sdk.Context, ticker types.Ticker) error {
+	// Check if Ticker already exists for the provider
+	alreadyExists, err := k.tickers.Has(ctx, types.TickerString(ticker.String()))
+	if err != nil {
+		return err
+	}
+	if alreadyExists {
+		return types.NewTickerAlreadyExistsError(types.TickerString(ticker.String()))
+	}
+	// Create the config
+	return k.tickers.Set(ctx, types.TickerString(ticker.String()), ticker)
+}
+
+// GetAllProvidersMap returns the set of Providers objects currently stored in state
+// as a map[TickerString] -> Providers.
+func (k *Keeper) GetAllProvidersMap(ctx sdk.Context) (map[string]types.Providers, error) {
+	iter, err := k.providers.Iterate(ctx, nil)
+	if err != nil {
+		return nil, err
+	}
+	keyValues, err := iter.KeyValues()
+	if err != nil {
+		return nil, err
+	}
+
+	m := make(map[string]types.Providers, len(keyValues))
+	for _, keyValue := range keyValues {
+		m[string(keyValue.Key)] = keyValue.Value
+	}
+
+	return m, nil
+}
+
+// CreateProviders initializes a new providers.
+// The Ticker.String corresponds to a market, and must be unique.
+func (k *Keeper) CreateProviders(ctx sdk.Context, providers types.Providers, ticker types.Ticker) error {
 	// Check if MarketConfig already exists for the provider
-	alreadyExists, err := k.markets.Has(ctx, types.TickerString(ticker.String()))
+	alreadyExists, err := k.providers.Has(ctx, types.TickerString(ticker.String()))
 	if err != nil {
 		return err
 	}
 	if alreadyExists {
 		return types.NewTickerAlreadyExistsError(types.TickerString(ticker.String()))
 	}
 	// Create the config
-	err = k.markets.Set(ctx, types.TickerString(ticker.String()), ticker)
+	return k.providers.Set(ctx, types.TickerString(ticker.String()), providers)
+}
+
+// GetAllPathsMap returns the set of Paths objects currently stored in state
+// as a map[TickerString] -> Paths.
+func (k *Keeper) GetAllPathsMap(ctx sdk.Context) (map[string]types.Paths, error) {
+	iter, err := k.paths.Iterate(ctx, nil)
+	if err != nil {
+		return nil, err
+	}
+	keyValues, err := iter.KeyValues()
+	if err != nil {
+		return nil, err
+	}
+
+	m := make(map[string]types.Paths, len(keyValues))
+	for _, keyValue := range keyValues {
+		m[string(keyValue.Key)] = keyValue.Value
+	}
+
+	return m, nil
+}
+
+// CreatePaths initializes a new Paths.
+// The Ticker.String corresponds to a market, and must be unique.
+func (k *Keeper) CreatePaths(ctx sdk.Context, paths types.Paths, ticker types.Ticker) error {
+	// Check if MarketConfig already exists for the provider
+	alreadyExists, err := k.paths.Has(ctx, types.TickerString(ticker.String()))
 	if err != nil {
 		return err
 	}
+	if alreadyExists {
+		return types.NewTickerAlreadyExistsError(types.TickerString(ticker.String()))
+	}
+	// Create the config
+	return k.paths.Set(ctx, types.TickerString(ticker.String()), paths)
```

Comment:
Should this do a basic check to make sure that the tickers supplied in a path are in the main set of tickers, similar to what is done in validate basic on the marketmap?

Reply:
Not critical rn, can be added later.